ims/common/bmi_config.py

```diff
@@ -0,0 +1,43 @@
+import ims.common.constants as constants
+
+
+def parse_config(cfg):
+    # Mandatory Options
+    # BMI Section
+    cfg.option(constants.BMI_SECTION, constants.UID_OPT)
+    cfg.option(constants.BMI_SECTION, constants.SERVICE_OPT)
+
+    # DB Section
+    cfg.option(constants.DB_SECTION, constants.DB_PATH_OPT)
+
+    # RPC Section
+    cfg.option(constants.RPC_SECTION, constants.RPC_SERVER_IP_OPT)
+    cfg.option(constants.RPC_SECTION, constants.RPC_SERVER_PORT_OPT, int)
+    cfg.option(constants.RPC_SECTION, constants.NAME_SERVER_IP_OPT)
+    cfg.option(constants.RPC_SECTION, constants.NAME_SERVER_PORT_OPT, int)
+
+    # TFTP Section
+    cfg.option(constants.TFTP_SECTION, constants.PXELINUX_PATH_OPT)
+    cfg.option(constants.TFTP_SECTION, constants.IPXE_PATH_OPT)
+
+    # REST API Section
+    cfg.option(constants.REST_API_SECTION, constants.REST_API_IP_OPT)
+    cfg.option(constants.REST_API_SECTION, constants.REST_API_PORT_OPT, int)
+
+    # Logs Section
+    cfg.option(constants.LOGS_SECTION, constants.LOGS_PATH_OPT)
+    cfg.option(constants.LOGS_SECTION, constants.LOGS_DEBUG_OPT, bool)
+    cfg.option(constants.LOGS_SECTION, constants.LOGS_VERBOSE_OPT, bool)
```

Comment:
Yeah, if we don't use keyword arg, then we risk being stuck to a certain library API with little flexibility in the future.

That's true, we will exceed the line length shown below and it may look a little less prettier but I don't think we should lose API flexibility for that. We're eventually going to be adding more options on a case by case basis. Like one idea was to validate if the IP address/Port in the config file was reachable. Another was we could validate if the Ceph Pool itself existed. If any of these fail, we throw a log.ERROR that some of the HW resources are not reachable and possibly send a notification to the administrator of BMI. This is up for discussion, would like more reviews on it but I'm digressing :P . Basically, API flexibility over prettiness for me though.  

cfg.option(constants.LOGS_SECTION, constants.LOGS_VERBOSE_OPT,
                 type=bool)